domain/userDB.go

```diff
@@ -0,0 +1,17 @@
+package domain 
+
+//UserDB represents a user database with file pointers and segments based on search results
+type UserDB struct {
+	//Document DB pointer
+    FilePointers [] string
+
+	//Vector search segments
+	Segments [] string
+}
+
+func NewUserDB(filePointers [] string, segments [] string) *UserDB {
```

Comment:
This does not pass by a slice pointer, instead in go it will pass by value the slice (though this will still point to the underlying original size) which is likely not intentional as certain operations like `append` will not be accessible outside of the calling function. https://medium.com/@ansujain/understanding-slices-in-go-pass-by-value-and-pass-by-pointer-9a52830c741e


```suggestion
func NewUserDB(filePointers *[]string, segments *[]string) *UserDB {
```

Reply:
Got it - like C passing value by reference

Reply:
So question:  If we are to pass as a cop the New method will create a new cop and hold it as part of the returned UserDB. Thinking about it a constructor in java. I don't want the caller to have any reference of underlying objects in UserDB. Is this best way to go about it? Is there something else that GO syntax or programming best practice?

Reply:
In this example, if you pass a "copy," it passes the memory locations of the original items in the slice, but some items, like the original calling slice's length and capacity, are not modified. If you want to learn more I recommend trying out the tutorials of [A Tour Of Go](https://go.dev/tour/list).

Example:

![image](https://github.com/user-attachments/assets/fed1900c-6ef8-46a9-9b3a-1cb7029d1ef2)

Reply:
Fixed